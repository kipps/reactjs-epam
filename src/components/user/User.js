import {connect} from 'react-redux';
import React from "react";

const User = (props) => {
  const { name } = props;
  return (
    <div className='User'>
      <a href='#' className='User__link black'>
          {name}
      </a>
    </div>
  );
}

const mapStateToProps = state => {
    return {
        name: state.userState.name,
    }
}

export default connect(mapStateToProps)(User)