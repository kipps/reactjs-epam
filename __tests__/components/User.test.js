import React from 'react';
import { shallow, render } from 'enzyme/build';
import { configure } from 'enzyme/build';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16/build';

import User from '../../src/components/user/User';


configure({ adapter: new Adapter() });

describe("User ", () => {
  it("renders default user", () =>
    expect(
      shallow(<User />)
        .find('div.User')
        .length
    ).toBe(1));

  it('user snapshot', () => {
    const tree = renderer
      .create(<User />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});