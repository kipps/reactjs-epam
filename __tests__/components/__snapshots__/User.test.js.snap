// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User  user snapshot 1`] = `
<div
  className="User"
>
  <a
    className="User__link black"
    href="#"
  />
</div>
`;
